dart-atom/dartlang
# Add Configuration Options For Max Warnings Per File and Project

> Proposed solution for https://github.com/dart-atom/dartlang/issues/940.

## How It Was Fixed:
- Add configuration  options for `maxIssuesPerFile` and `maxIssuesPerProject`.
  - Had to bump the `order` of the options after that.
- Change the `const` variables `_maxIssuesPerFile` and `_maxIssuesPerProject` to be getters.
  - They can no longer be `const` because they can be updated.
  - Use the values from the configuration options in the getters.

## Testing Suggestions:
- Pull down this branch:
  - See that you can change the max issues displayed per project and file.
  - Change them.
  - Run `Reanalyze Sources`
  - Create enough errors that they get truncated by the max values.
  - See that get get truncated at the provided values.

FYA: @devoncarew
